en/drafts/originals/agent-based-model-communication-networks.md

```diff
@@ -555,22 +559,20 @@ plt.colorbar(label="Number of Agents present in Cell")
 
 ```
 
-{% include figure.html filename="en-or-agent-based-model-communication-networks-04.png" alt="This shows a color mesh, a 2-dimensional grid where each cell in the grid is colored based on how many agents are present on it, on a scale of 0 to 3, with lighter yellow colors indicating more agents and darker blue colors indicating less agents. The figure shows that some cells are more crowded than others, but the distribution does look rather random." caption="Figure 4. A color mesh showing how many agents are present on each cell of our grid space" %}
+{% include figure.html filename="en-or-agent-based-model-communication-networks-04.png" alt="Two-dimensional grid where each cell is colored based on how many agents are present on it, on a scale of 0 to 3, with lighter yellow colors indicating more agents and darker blue colors indicating fewer agents." caption="Figure 4. Color mesh showing the number of agents present on each cell of our grid space" %}
 
 > _Bonus question 3_:
-Letters are sent to direct neighbors. How could you implement sending letters only to agents far apart, e.g., with at least a distance of three cells? *Hint*: You will have to define a distance measure on grids, see e.g., this [tutorial on similarity measures](https://programminghistorian.org/en/lessons/common-similarity-measures#city-block-manhattan-distance).
+Letters are sent to direct neighbors. How could you implement sending letters only to agents far apart, e.g. with at least a distance of three cells? *Hint*: You will have to define a distance measure on grids. See for example this [_Programming Historian_ lesson on similarity measures](https://programminghistorian.org/en/lessons/common-similarity-measures#city-block-manhattan-distance).
```

Comment:
'Letters are sent to direct neighbours': I think you mean that an agent's direct neighbour is an agent who shares the same cell, but this is confusing because we've recently been speaking of cell neighbours as the cells right next to a given cell. 

Is there a different way we could phrase this question?

Reply:
```suggestion
Letters are sent to agents in neighboring cells. How could you implement sending letters only to agents further apart than that, e.g. with at least a distance of three cells? *Hint*: You will have to define a distance measure on grids. See for example this [_Programming Historian_ lesson on similarity measures](https://programminghistorian.org/en/lessons/common-similarity-measures#city-block-manhattan-distance).
```

Reply:
Hmm, I think I may not be understanding the agents' step method correctly then. I was under the impression that up until now, agents would 1. move to a neighbouring cell, then 2. send a letter to an agent that shared that cell.

(From these sentences:
"Now we need to add to the agents’ behaviors, letting them move around and only send letters to other agents in the same cell. [...] First, let’s handle movement, and have the agents move to a neighboring cell. [...] Next, we need to get all the other agents present in a cell, and send one of them a letter.")


Or perhaps by "letters are sent to agents in neighbouring cells" (in the Bonus Question here) you mean that those agents were in the _previously_ neighbouring cell, i.e. before the sender moved to that cell? If that's the case, I can work around it to make that clearer.

Apologies – I just want to make sure I'm not introducing any errors to the lesson!

Reply:
Oof, of course you're right, I'm sorry. That's what I get from suggesting things while in a meeting...

```suggestion
Letters are sent to direct neighbors inside the same cell. How could you implement sending letters only to agents further apart, e.g. with at least a distance of three cells? *Hint*: You will have to define a distance measure on grids. See for example this [_Programming Historian_ lesson on similarity measures](https://programminghistorian.org/en/lessons/common-similarity-measures#city-block-manhattan-distance).
```

Reply:
Thank you, that's very helpful!